### PR TITLE
mod+itest: reproduce and fix reverse direction invoice payment, bump to rc4

### PR DESCRIPTION
**Pull request by guggero**
Reported offline by @jamaljsr.

Depends on https://github.com/lightningnetwork/lnd/pull/8908.
Depends on https://github.com/lightninglabs/taproot-assets/pull/1011.

Fixes a bug that when an invoice is paid, the full invoice amount is checked when validating the local channel balance and reserve. But for custom channels the actual HTLC amount might be just above dust and shouldn't validate the channel reserve.
We update our itest to reproduce the issue and then update to the lnd commit that fixes it.

### CODE REVIEW

File: itest/litd_custom_channels_test.go [hunk 1]
```diff
@@ -227,13 +227,13 @@ func testCustomChannels(_ context.Context, net *NetworkHarness,
...
 	// there is enough on-chain balance on Dave's side to be able to create
-	// an HTLC.
-	sendAssetKeySendPayment(
-		t.t, dave, charlie, keySendAmount, assetID, fn.None[int64](),
+	// an HTLC. We use an invoice to execute another code path.
+	invoiceResp := createAssetInvoice(
+		t.t, dave, charlie, keySendAmount, assetID,
 	)
-	logBalance(t.t, nodes, assetID, "after keysend back")
+	payInvoiceWithAssets(t.t, dave, charlie, invoiceResp, assetID)
```
Comment (GeorgeTsagk) [inline itest/litd_custom_channels_test.go:236]: so prior to fix the sender would crash here saying there's no available path?

**Review by GeorgeTsagk — APPROVED** [review]
Diff LGTM
itest seems unhappy, doesn't look related to these changes
https://github.com/lightninglabs/lightning-terminal/actions/runs/9890241755/job/27318070143?pr=790#step:6:1335

File: itest/litd_custom_channels_test.go (hunk 1 above)
Comment (guggero) [reply itest/litd_custom_channels_test.go:236]: Yes, a NO_ROUTE error would be thrown here.

**Comment by guggero** [comment]
>  itest seems unhappy, doesn't look related to these changes

Oops, I had a change in the flag name in there that isn't referenced yet. Will need to update `tapd` once this is merged: https://github.com/lightninglabs/taproot-assets/pull/1011 (which will then require the rename of the mock price flag).

**Comment by guggero** [comment]
> I assume I should have expected this version of litd to not recognize the config option unknown option: `taproot-assets.experimental.rfq.mockoraclecentpersat`?

Sorry, I forgot to mention that. Yes, we renamed the config flag because its name was actually wrong.
Can you update your guides/documents accordingly please?
There's now also an alternative option: `taproot-assets.experimental.rfq.mockoraclesatsperasset` which allows you to specify the mock exchange rate in a different way.

**Comment by jamaljsr** [comment]
A direct peer payment in the opposite direction now works, but when I perform a multi-hop payment in the opposite direction, it results in a `FAILURE_REASON_NO_ROUTE` error. 

I have setup a regtest network with the following setup. 
```
alice --[asset]-> bob --[sats]-> carol --[asset]-> dave
```

I performed the following steps:
1. Mint 1000 PBUX on alice
2. Send 500 PBUX from alice to dave
3. Open 400 PBUX channel from alice to bob
4. Open 400 PBUX channel from carol to dave
5. Open 500K sats channel from bob to carol
6. Pay a 200K sat invoice from bob to carol
7. Pay a 200 PBUX invoice from alice to dave (success)
8. Pay a 100PBUX invoice from dave to alice (`FAILURE_REASON_NO_ROUTE`)

Here are the logs for all of the nodes.
[docker-logs.zip](https://github.com/user-attachments/files/16180950/docker-logs.zip)

**Comment by jamaljsr** [comment]
When I try to make a large direct peer invoice payment, it results in the litd process hanging and needing to be killed. It won't shutdown gracefully.

This regtest network just has 2 litd nodes. I perform these steps to reproduce the error.
1. Mint 1,000,000 PBUX on alice
2. Open 500,000 PBUX channel from alice to bob
3. Create a 10,000 PBUX invoice on bob
4. Paying the invoice on alice hangs the litd node

Here are the logs for both nodes.
[alice.log](https://github.com/user-attachments/files/16181111/alice.log)
[bob.log](https://github.com/user-attachments/files/16181112/bob.log)

**Comment by guggero** [comment]
The first issue above (https://github.com/lightninglabs/lightning-terminal/pull/790#issuecomment-2223415497) is fixed with the latest push.

The second issue (https://github.com/lightninglabs/lightning-terminal/pull/790#issuecomment-2223435601) is likely a bit more involved, so I spun it out into a separate issue report: https://github.com/lightninglabs/taproot-assets/issues/1013

**Comment by Liongrass** [comment]
I am running this now, litd started up promptly and seems to be running smoothly!

**Review by jamaljsr — APPROVED** [review]
> The first issue above ([#790 (comment)](https://github.com/lightninglabs/lightning-terminal/pull/790#issuecomment-2223415497)) is fixed with the latest push.

tACK 🔥 Confirmed the first issue is resolved now. Thank you for the quick fix 💪

**Review by GeorgeTsagk — APPROVED** [review]
re-ACK

**Pull request merged**